### ISSUE TITLE
Future.task should be able to take arguments

### ISSUE THREAD

**Issue by fresheneesz** [laverdet/node-fibers#215]
Itd be really nice if you could do something like:

```
something.on("someEvent", Future.task(function(event) {
   // do stuff with event
})
```

But the return value of Future.task doesn't take in arguments. Could it be changed to be able to take arguments?

**Comment by laverdet** [comment]
What you want is `.future()` [installed on Function.prototype]. So:

```js
something.on("someEvent", function(event) {
   // do stuff with event
}.future())
```

The only problem is that your exceptions will be gobbled up because Node doesn't `wait` or `detach` your future.

I'd recommend this instead:
```js
something.on("someEvent", function(event) {
  Future.task(function() {
    // do stuff with event
  }).detach();
})
```

Perhaps I could an extra argument to `future` to detach it (throwing exceptions back into the Node event loop)

**Comment by fresheneesz** [comment]
@laverdet I must say, the difference between `function.future()` and `Future.task()` is still pretty vague to me. But thanks for the suggestions. 

My last comment here might be relevant to this, since its a new proposal on how to get rid of the need for detach entirely: https://github.com/laverdet/node-fibers/pull/115

**Comment by laverdet** [comment]
`.future()` returns a function which, when called, will return a future. It does not actually call the function, in this way it is similar to `bind`

`Future.task()` calls a function and immediately returns a future. In this way it is similar to `apply` or `call`.

**Comment by fresheneesz** [comment]
Right, but is `(function(){...}.future())()` the same thing as `Future.task(function(){...})` ? Also, would love to hear your thoughts on getting rid of the need for detach.

**Comment by laverdet** [comment]
Yeah, they're the same. Future.task is implemented just that way:

```js
Future.task = function(fn) {
  return fn.future()();
}
```
[actually not totally this way, Future.task has the ability to run multiple parallel tasks, check the source if you're curious]

Getting rid of `detach` would require a pretty large retooling of Future and code that's already using it. For instance this is not uncommon at all:

```js
var future1 = SomeAPI();
var future2 = SomeOtherAPI();
return future1.wait() + future2.wait();
```

Bluebird's solution works because all error handlers are supplied up front. In fibers we call `wait` on `future1` first (which will probably yield to the event loop) and then we'll call `wait` on `future2` (which may or may not actually yield depending on which finished first internally). If it turns out that `future1` actually throws then we'll never even call `future2.wait()`. Trying to detect this kind of thing gets *really* hairy and is going to yield more harm than good.

The best solution here is just a naming convention in your libraries. All functions which return a future should have "future" in their name. This is similar to the reasoning behind the * notation in generators.

**Comment by fresheneesz** [comment]
Ugh, I hate when you're always right.
